state/src/replicationv2/mod.rs

```diff
@@ -81,4 +230,48 @@ mod test {
     fn test_openraft_suite() {
         openraft::testing::Suite::test_all(StorageBuilder).unwrap();
     }
+
+    #[tokio::test]
+    async fn test_raft_singleton() {
+        let node = MockRaft::create_singleton_raft().await;
+        let wallet = mock_empty_wallet();
+
+        // Propose a new wallet to the raft
+        let id = wallet.wallet_id;
+        node.get_raft()
+            .client_write(Box::new(StateTransition::AddWallet { wallet }))
+            .await
```

Comment:
this is nice, we can remove `ProposalWaiter`s then, right?

Reply:
Not quite, going to add back in proposal waiters afterward to allow for fire and forget semantics if a client doesn't want to await a transition